### PR TITLE
Added a new angular project for the digital library.  Added Chromium …

### PR DESCRIPTION
**Pull request by bgsavage**
…headless browser for running unit test.  Added a DataService and MockDataService which replaces it when the app is built with the standalone configuration.

Most of this code is scaffolded via `ng new`, but I'll call out parts that I've configured and added.

### CODE REVIEW

File: webapp/src/app/data/data.module.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import { DataService } from 'src/app/data/data.service';
+import { HttpClientModule } from '@angular/common/http';
+
+@NgModule({
+  declarations: [ ],
+  imports: [
+    CommonModule, 
+    HttpClientModule
+  ],
+  providers: [{ provide: DataService, useClass: environment.dataService }]
```
Comment (bgsavage) [inline webapp/src/app/data/data.module.ts]: Added this DataModule.  Here the DataService provider is defined by the environment.

File: webapp/src/app/data/data.service.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * Wrapper service for Angular's HttpClient.  As a best practice, we shouldn't ever use
+ * HttpClient directly except for here, so that any future Angular HttpClient API changes
+ * can be updated to this one spot.  This also allows us the opportunity add any generic,
+ * Sbdl-specific logic in a single spot that should be applied across the board.
+ */
+@Injectable()
+export class DataService {
+
+  // TODO: Set and use CDL environment variable.
+  constructor(private httpService: HttpClient) { }
```
Comment (bgsavage) [inline webapp/src/app/data/data.service.ts]: I never like to call `HttpClient` directly in my service classes for a few reasons.  This is just a generic DataService class I always include which wraps `HttpClient`.  For this project, I imagine we will handle environment configurations to call the external CDL api url, and provide a different different implementation of this class when running locally in a standalone mode.

File: webapp/src/environments/environment.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { DataService } from 'src/app/data/data.service';
+
+// This file can be replaced during build by using the `fileReplacements` array.
+// `ng build --prod` replaces `environment.ts` with `environment.prod.ts`.
+// The list of file replacements can be found in `angular.json`.
+
+export const environment = {
+  production: false,
+  standalone: false,
+  dataService: DataService
```
Comment (bgsavage) [inline webapp/src/environments/environment.ts]: For environment.prod and the default environment, use the live DataService will will make external api calls.

File: webapp/src/environments/environment.standalone.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { MockDataService } from 'src/app/data/mock-data.service';
+
+export const environment = {
+    production: false,
+    standalone: true,
+    dataService: MockDataService
```
Comment (bgsavage) [inline webapp/src/environments/environment.standalone.ts]: For standalone environment, use the MockDataService instead.

File: webapp/package.json [hunk 5]
```diff
@@ -0,0 +1,50 @@
+{
+  "name": "sb-digital-library",
+  "version": "0.0.0",
+  "scripts": {
+    "ng": "ng",
+    "start": "npm run test && ng serve",
+    "start-standalone": "npm run test && ng serve --c standalone",
+    "build": "npm run test && ng build --prod --aot --vendor-chunk",
+    "build-standalone": "npm run test && ng build --c standalone --aot --vendor-chunk",
```
Comment (bgsavage) [inline webapp/package.json]: Custom start and build commands will run the unit tests by default.

File: webapp/angular.json [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    "sb-digital-library": {
+      "projectType": "application",
+      "schematics": {
+        "@schematics/angular:component": {
+          "style": "scss"
+        }
+      },
+      "root": "",
+      "sourceRoot": "src",
+      "prefix": "sbdl",
```
Comment (bgsavage) [inline webapp/angular.json]: Defined the app prefix to be `sbdl`. Thoughts?

File: webapp/angular.json [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+              ]
+            },
+            "standalone" : {
+              "fileReplacements": [
+                {
+                  "replace": "src/environments/environment.ts",
+                  "with": "src/environments/environment.standalone.ts"
+                }
+              ]
+            }
```
Comment (bgsavage) [inline webapp/angular.json]: Standalone configuration

File: webapp/angular.json (hunk 6 above)
Comment (jdbernard) [reply webapp/angular.json]: Works for me. 👍

File: webapp/src/app/data/data.service.ts (hunk 2 above)
Comment (jdbernard) [reply webapp/src/app/data/data.service.ts]: I tend to do the same. I agree with the approach of swapping in different implementations of the `DataService` class depending on which mode we're in.

**Pull request merged**